mtags/src/main/scala/scala/meta/internal/pc/CompilerAccess.scala

```diff
@@ -149,15 +170,45 @@ class CompilerAccess(
   private def handleError(e: Throwable): Unit = {
     CompilerThrowable.trimStackTrace(e)
     logger.log(Level.SEVERE, e.getMessage, e)
-    shutdown()
+    restart()
   }
   private var _compiler: MetalsGlobal = _
-  private val lock = new Object
   private def loadCompiler(): MetalsGlobal = {
     if (_compiler == null) {
       _compiler = newCompiler()
     }
     _compiler.reporter.reset()
     _compiler
   }
+
+  private def onCompilerJobQueue[T](
+      thunk: () => T,
+      token: CancelToken
+  ): CompletableFuture[T] = {
+    val result = new CompletableFuture[T]()
+    jobs.submit(result, { () =>
```

Comment:
I feel that this should be somehow abstracted away - but I have no idea how :)

Reply:
I removed the `Thread.interrupted()` from `withInterruptableCompiler` since it was redundant. Otherwise, I don't think there's a need to abstract over anything here.